hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -164,11 +166,29 @@ def get_signature(target: Any, *, follow_wrapped: bool = True) -> inspect.Signat
             return sig.replace(
                 parameters=[v for k, v in sig.parameters.items() if k != "self"]
             )
-    sig = inspect.signature(target, follow_wrapped=follow_wrapped)
+    sig = _inspect_signature(target, follow_wrapped=follow_wrapped, eval_str=eval_str)
```

Comment:
```suggestion
    kw = {"eval_str": eval_str} if sys.version_info[:2] >= (3, 10) else {}
    sig = _inspect_signature(target, follow_wrapped=follow_wrapped, **kw)
```
and then we don't need the definitions below.